src/main/scala/com/typesafe/sbt/site/Compat.scala

```diff
@@ -49,12 +49,12 @@ object Compat {
     }
   }
 
-  def watchSettings(config: Configuration): Seq[Setting[_]] =
+  def watchSettings(scope: Scope): Seq[Setting[_]] =
     Seq(
       ConfigGlobal / watchSources += new Source(
-        base = (sourceDirectory in config).value,
-        includeFilter = (includeFilter in config).value,
-        excludeFilter = (excludeFilter in config).value
+        base = (sourceDirectory in scope).value,
+        includeFilter = (includeFilter in scope).value,
+        excludeFilter = (excludeFilter in scope).value
```

Comment:
I made it more general, so I could add watchsettings not only under Configuration scope, but also under a task scope.